[INDICATORS] Neopixel 2leds

mrickma opened the pull request:
## Description:  Addition of the Adafruit NeoPixel library to replace FastLED
The Adafruit NeoPixel library is used as a full alternative to FastLED to address RGB LEDs. This solves the problem that FastLED results in nearly immediate reboots of ESP32C3 boards.
The request also includes new [env:esp32c3-dev-c2-ble] environments and a change to [env:lolin_c3_mini] for the boards on which the addition was developed.
coauthor Argafal <dev.omg@argafal.de>
## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

mrickma committed:
Add Adafruit NeoPixel library #defines

* Currently FastLED causes rapid reboots on ESP32C3 boards when OMG signals.
  Onboard RGB LEDs can be adressed more reliably with the NeoPixel library.
  Define RGB_INDICATORS=2 to use this addition.
* rebase fix: adapt to "Bump ESP32" changes in development

mrickma committed:
Add build environments for the Espressiv ESP32-C3-DevKitC-02 board

* esp32c3-dev-c2-ble and esp32c3-dev-c2-ble-no-serial environments both
  use the Adafruit NeoPixel library for the onboard RGB LED as FastLED
  causes rapid reboots.

* Only occasional reboots of this board occure when serial output is not
  read. The -no-serial version further improves stability.

mrickma committed:
Add a build environment for the Espressiv ESP32-C3-DevKitC-02 board

* The esp32c3-dev-c2-ble offers a reasonable compromise between logging
  and stability of OMG on this board.

mrickma committed:
Naming mess up of environments fixed

mrickma committed:
Add comment for RGB_INDICATORS=2, thanks argafal

mrickma committed:
Change the build environment for the Wemos Lolin C3 mini v2.1.0 board

* The env:lolin_c3_mini now uses the Adafruit NeoPixel library for the
  onboard RGB LED.
  authored by Argafal <dev.omg@argafal.de>

mrickma committed:
Fixed really bad typo, missing single quote

mrickma committed:
Update to use Adafruit NeoPixel color signalling schemes

* build for blue based signalling colors
  add hint how to change brightness

mrickma committed:
Added color combinations for good visibility of OMG signalling

* Default is Argafal's green based scheme.
  Use -DANEOPIX_COLOR_SCHEME=1 build for the gold on blue variant.
* Added color to booting to better recognize reboots

mrickma committed:
Argafal added comment line to env:lolin_c3_mini_with_serial

mrickma committed:
Argafal changed colors in the green based scheme

* Now only the color for OMG info signal is different between schemes

mrickma committed:
Stripped down to the essentials

* Unused colors were removed and the code was restructured in parts
  shared by the schemes and differences.

mrickma committed:
Removed definition and usage of ANEOPIX_BOOT_LED

* If it had been defined as separate LED it would have never been
  turned off. Boot signalling is done by the info LED now.

mrickma committed:
Use ${com.esp32_platform} in esp32c3-dev-c2-ble environment

mrickma committed:
OpenMQTTGateway signals complete by Adafruit NeoPixel

* Added code for ANEOPIX_IND_DATA_GPIO2, the second separate LED.
* Extended colors and color schemes. The original FastLED colors
  are default now. The green and blue based combinations were
  shifted up.
* In the environments existing ANEOPIX_COLOR_SCHEME definitions are +1 now.

mrickma committed:
Integrate the Adafruit NeoPixel library into OpenMQTTGateway's
platformio.ini build system

* esp32c3-dev-c2-ble and lolin_c3_mini environments changed accordingly

mrickma committed:
rebase fix: simplify the lolin_c3_mini environment

1technophile commented:
Thanks for the PR. For the lint error, you can right-click in the code from VSC and click on Format document. 
It will adapt the code to the required formatting.

mrickma committed:
lint fix: reformat in VS code

1technophile commented on main/User_config.h:
```diff
@@ -497,7 +497,132 @@ int lowpowermode = DEFAULT_LOW_POWER_MODE;
+#  define SetupIndicatorInfo()
+#  define SetupIndicatorSendReceive()
+#  define SetupIndicatorError()
+#else // FastLED remains standard for RGB_INDICATORS
```
I think we could completely replace the use of FASTLED for indicators, and remove the code below, thoughts ?

mrickma replied:
I think so too! I was just afraid of a big step. It would require a rather simple find/replace action in environments.ini. Shall I make a suggestion?

1technophile replied:
Yes go ahead

mrickma committed:
Completely replace FastLED by Adafruit_NeoPixel for indicators

* Remove FastLED code from main/User_config.h
  Rename ANEOPIX_LED_TYPE to ANEOPIX_IND_TYPE
* Adapted in environments.ini:
  env:esp32-m5atom-matrix, env:esp32-m5atom-lite, env:esp32s3-dev-c1-ble,
  env:esp32c3-dev-m1-ble, env:esp32c3-dev-c2-ble, env:lolin_c3_mini,
  env:thingpulse-espgateway, env:thingpulse-espgateway-ethernet
* The APA102 led of env:tinypico-ble could not be addressed. This env
  was missing the -DRGB_INDICATORS=true directive anyhow.

1technophile commented on environments.ini:
```diff
@@ -895,16 +894,16 @@ lib_deps =
-  '-DFASTLED_IND_DATA_GPIO=2,12,RGB'
-  '-DFASTLED_IND_TYPE=APA102'
-  '-DFASTLED_IND_NUM_LEDS=1'
+;  '-DRGB_LED_POWER=13'
+;  '-DFASTLED_IND_DATA_GPIO=2,12,RGB'
+;  '-DFASTLED_IND_TYPE=APA102'
+;  '-DFASTLED_IND_NUM_LEDS=1'
```
you can remove this

1technophile commented on main/User_config.h:
```diff
@@ -498,78 +498,125 @@ int lowpowermode = DEFAULT_LOW_POWER_MODE;
+#  ifndef ANEOPIX_INFO_LED
+#    define ANEOPIX_INFO_LED 0 // First Led
+#  endif
+#  ifndef ANEOPIX_SEND_RECEIVCE_LED
+#    define ANEOPIX_SEND_RECEIVCE_LED 0 // First Led
```
Do you mind taking the opportunity of this PR to fix `RECEIVE`

mrickma committed:
Adjustments

* Delete unused FastLED directives from env:tinypico-ble
* Fix typo in RECEIVE

1technophile commented on main/User_config.h:
```diff
@@ -498,78 +498,125 @@ int lowpowermode = DEFAULT_LOW_POWER_MODE;
+Adafruit_NeoPixel leds(ANEOPIX_IND_NUM_LEDS, ANEOPIX_IND_DATA_GPIO, ANEOPIX_IND_TYPE);
+#  ifdef ANEOPIX_IND_DATA_GPIO2 // Only used for Critical Indicator
+// assume the same LED type
+Adafruit_NeoPixel leds2(ANEOPIX_IND_NUM_LEDS, ANEOPIX_IND_DATA_GPIO2, ANEOPIX_LED_TYPE);
```
I missed this, could you change `ANEOPIX_LED_TYPE` to `ANEOPIX_IND_TYPE`

mrickma committed:
fixed forgotten ANEOPIX_IND_TYPE

1technophile commented:
Thanks
